packages/core/contracts/financial-templates/long-short-pair/LongShortPair.sol

```diff
@@ -223,23 +224,20 @@ contract LongShortPair is Testable, Lockable {
      */
     function settle(uint256 longTokensToRedeem, uint256 shortTokensToRedeem)
         public
-        postExpiration()
         nonReentrant()
         returns (uint256 collateralReturned)
     {
-        // If the contract state is open and postExpiration passed then `expire()` has not yet been called.
-        require(contractState != ContractState.Open, "Unexpired contract");
-
-        // Get the current settlement price and store it. If it is not resolved, will revert.
-        if (contractState != ContractState.ExpiredPriceReceived) {
-            expiryPrice = _getOraclePriceExpiration(expirationTimestamp);
-            // Cap the return value at 1.
-            expiryPercentLong = Math.min(
-                financialProductLibrary.percentageLongCollateralAtExpiry(expiryPrice),
-                FixedPoint.fromUnscaledUint(1).rawValue
-            );
-            contractState = ContractState.ExpiredPriceReceived;
-        }
+        // Either time must have passed the expirationTimestamp (normal expiration) or it is before expirationTimestamp
+        // and early expiration is enabled enabled.
+        require(
+            getCurrentTime() > expirationTimestamp || (getCurrentTime() < expirationTimestamp && enableEarlyExpiration),
+            "Can not settle"
+        );
+
+        // Get the settlement price and store it. Also sets expiryPercentLong to inform settlement. Reverts if either:
+        // a) the price request has not resolved (either a normal expiration call or early expiration call) or b) If the
+        // the contract was attempted to be settled early but the price returned is the ignore oracle price.
+        if (!receivedSettlementPrice) getExpirationPrice();
```

Comment:
Why can't you check `expiryPrice == 0`?

Reply:
Ah

Reply:
Can you add this comment above `receivedSettlementPrice`? Otherwise it seems redundant at first, until you realize 0 is a valid price.